src/porepy/numerics/ad/operators.py

```diff
@@ -1295,7 +1295,7 @@ def parse(self, mdg: pp.MixedDimensionalGrid) -> np.ndarray:
         return self._values
 
 
-class TimeDependentDenseArray(DenseArray):
+class TimeDependentDenseArray(Operator):
```

Comment:
Please explain!

Reply:
I modified some things inside TimeDependentDenseArray, and noticed that it repeats several things from the operator initialization, so I decided to clean it up a bit. 

As I understand it, the point why we inherit from DenseArray is that we might use a check `isinstance(..., DenseArray)`, and want to catch TimeDependentDenseArrays as well. However, I don't see when this can be actually used, because DenseArray and TimeDependentDenseArrays have completely different attributes, so if one wants to do something specific only if the operator is a DenseArray, they will anyway have to add one more check `isinstance(..., TimeDependentDenseArray)` to treat them differently. Tell me if I didn't get it right, please.

Reply:
I agree. @keileg?

Reply:
Makes sense.